lib/Controller/ActionStepController.php

```diff
@@ -0,0 +1,123 @@
+
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Mail\Controller;
+
+use OCA\Mail\AppInfo\Application;
+use OCA\Mail\Exception\ServiceException;
+use OCA\Mail\Http\JsonResponse;
+use OCA\Mail\Http\TrapError;
+use OCA\Mail\Service\ActionStepService;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\Attribute\NoAdminRequired;
+use OCP\IRequest;
+
+class ActionStepController extends Controller {
+	private ?string $uid;
+
+	public function __construct(
+		IRequest $request,
+		?string $userId,
+		private ActionStepService $actionStepService,
+	) {
+		parent::__construct(Application::APP_ID, $request);
+		$this->uid = $userId;
+	}
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * @return JsonResponse
+	 */
+	#[TrapError]
+	public function findAllStepsForAction(int $actionId): JsonResponse {
+		if ($this->uid === null) {
+			return JsonResponse::error('User not found', Http::STATUS_UNAUTHORIZED);
+		}
+		$actionSteps = $this->actionStepService->findAll($actionId);
```

Comment:
pass the user ID too to prevent https://en.wikipedia.org/wiki/Insecure_direct_object_reference